File: soccer/src/main/java/com/jk/soccer/model/local/DBDao.java
package com.jk.soccer.model.local;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import com.jk.soccer.etc.enumeration.Type;

import java.util.List;

@Dao
public interface DBDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertSearch(List<TableSearch> entries);

    @Query("SELECT * FROM tableSearch WHERE Name LIKE :name AND Type = :type")
    LiveData<List<TableSearch>> getSearch(String name, Type type);

    @Query("SELECT * FROM tableSearch WHERE Type = :type AND ParentID = :parentID")
    List<TableSearch> getList(Type type, Integer parentID);

    @Query("DELETE FROM tableSearch WHERE ParentID != 0")
    void clearSearch();

}

File: price/src/main/java/com/jk/price/DBDao.kt
package com.jk.price

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface DBDao {
    @Insert
    suspend fun insert(purchase: Purchase)

    @Query("SELECT * FROM purchase " +
            "WHERE Market LIKE :market AND Type LIKE :type AND Name LIKE :name")
    fun search(market: String, type: String, name: String): LiveData<List<Purchase>>

    @Query("DELETE FROM purchase WHERE ID = :ID")
    suspend fun delete(ID: Int)
}